docs/developers/SubstraitModifications.md

```diff
@@ -0,0 +1,30 @@
+---
+layout: page
+title: Substrait Modifications
+nav_order: 4
+parent: Developer Overview
+---
+# Substrait Modifications in Gluten
```

Comment:
Do we need to mention the base version of Substrait?

Reply:
Added the base version.